mlir-sycl/include/mlir/Dialect/SYCL/Utils/Utils.h

```diff
@@ -40,6 +45,11 @@ SYCLAccessorSubscriptOp createSYCLAccessorSubscriptOp(AccessorPtrValue accessor,
                                                       OpBuilder builder,
                                                       Location loc);
 
+/// Populate \p wgSizes with workgroup size per dimensionality, given the rank
+/// \p numDims.
+void populateWorkGroupSize(SmallVectorImpl<Value> &wgSizes, unsigned numDims,
```

Comment:
Why here? This is a specialized function that make sense to have only in the transformation that uses `LoopInternalization`.

Reply:
It is useful to the transformation `LoopInternalization` but I do not think it belongs here.

Reply:
Ettore's other comment: https://github.com/intel/llvm/pull/9894#discussion_r1231354471:

> My opinion is that `populateWorkGroupSize` is a specialized function that is useful to the LoopInternalization pass. Not a generic utility function that would be useful to the SYCL dialect. 
>
> Perhaps other ppl will have a different opinion, but that is how I feel.

Reply:
I am going to let this PR merge in, given that the contentious matter is not critical. I will post a separate PR to move this function and `constructSYCLID` to LoopInternalization.